server/lib/nl/fulfillment/ranking.py

```diff
@@ -14,53 +14,73 @@
 
 from typing import List
 
+from lib.nl import utils
 from lib.nl.detection import ClassificationType
 from lib.nl.detection import ContainedInClassificationAttributes
 from lib.nl.detection import ContainedInPlaceType
 from lib.nl.detection import Place
 from lib.nl.detection import RankingClassificationAttributes
 from lib.nl.detection import RankingType
+from lib.nl.fulfillment import context
 from lib.nl.fulfillment.base import add_chart_to_utterance
 from lib.nl.fulfillment.base import ChartVars
 from lib.nl.fulfillment.base import populate_charts
 from lib.nl.fulfillment.base import PopulateState
-from lib.nl.fulfillment.context import classifications_of_type_from_context
 from lib.nl.utterance import ChartOriginType
 from lib.nl.utterance import ChartType
 from lib.nl.utterance import Utterance
 
 
+#
```

Comment:
I wonder if we should have two rankings fulfillment, which can eliminates the condition check and make it straightforward?

Reply:
I'm a bit divided on that.  Because the fork between ranking for places vs. ranking for svs is subtle and doing that upstream might mean we add more logic in common code.  In any case, Guha wants us to do multi-query-type handling, and as part of that we can reconsider?

Reply:
sgtm